Realm/RLMArray.mm

```diff
@@ -437,3 +438,23 @@ - (instancetype)reversedSortDescriptor {
 }
 
 @end
+
+@interface RLMArray (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMArray (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    @throw RLMException(@"Unexpected handover of unmanaged `RLMArray`");
```

Comment:
Could I define a macro `REALM_ASSERTION_FAILURE()` that unconditionally calls `REALM_UNREACHABLE()` after printing a message? It would seem odd to simply call unreachable here.